account_move_line_base_tax/account.py

```diff
@@ -86,27 +86,24 @@ class account_invoice_tax(osv.Model):
     _inherit = "account.invoice.tax"
 
     def move_line_get(self, cr, uid, invoice_id, context=None):
-        res = []
-        super(account_invoice_tax, self).move_line_get(cr, uid, invoice_id)
+        res = super(account_invoice_tax, self).move_line_get(
+            cr, uid, invoice_id, context=context)
         tax_invoice_ids = self.search(cr, uid, [
             ('invoice_id', '=', invoice_id)], context=context)
         for inv_t in self.browse(cr, uid, tax_invoice_ids, context=context):
             if not inv_t.base_amount and not inv_t.tax_code_id and not\
                     inv_t.tax_amount:
```

Comment:
Just to comment, when there are too many negative conditions, it could be good using

if not all([a, b, c, d]):
   statement

instead of

if not a and not b and not c and not d:

all() check that all the elements are true otherwise returns false,

conversely any() will return true whenever any of them is true, if all are false then
will return false.